engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -250,8 +257,19 @@ private[spel] class Typer(classLoader: ClassLoader, commonSupertypeFinder: Commo
           }
         case _ => invalid("Invalid ternary operator") // shouldn't happen
       }
-      //TODO: what should be here?
-      case e: TypeReference => fixed(Unknown)
+
+      case e: TypeReference => {
+
+        if (staticMethodInvocationsChecking) {
+
+          typeDefinitionSet.validateTypeReference(e, evaluationContextPreparer) match {
+            case Valid(typedClass: TypedClass) => valid(Unknown)
+            case Invalid(error) => Invalid(error)
```

Comment:
You have `.map` method for that. In the free time you can read about functors and monads. Basically functor has `map` method, and monad has `flatmap` and `pure`. But look out, because algebraic theory can blow your mind! ;)  Here are some useful content: https://eed3si9n.com/herding-cats/Functor.html and here are some nice pictures: https://adit.io/posts/2013-04-17-functors,_applicatives,_and_monads_in_pictures.html

Reply:
changed, but i also cast the Unknwon otherwise it didn't accept return type

Reply:
Changed it, it looks much better, thanks